uaclient/contract.py

```diff
@@ -183,32 +194,28 @@ def request_updated_contract(cfg, contract_token=None, allow_enable=False):
             'Got unexpected contract_token on an already attached machine')
     contract_client = UAContractClient(cfg)
     if contract_token:  # We are a mid ua-attach and need to get machinetoken
-        try:
-            new_token = contract_client.request_contract_machine_attach(
-                contract_token=contract_token)
-        except util.UrlError:
-            return False
+        new_token = contract_client.request_contract_machine_attach(
+            contract_token=contract_token)
     else:
         machine_token = orig_token['machineToken']
         contract_id = orig_token['machineTokenInfo']['contractInfo']['id']
+        new_token = contract_client.request_machine_token_refresh(
+            machine_token=machine_token, contract_id=contract_id)
+    errors = []
+    for name, entitlement in sorted(cfg.entitlements.items()):
+        if entitlement['entitlement'].get('entitled'):
+            # Obtain each entitlement's accessContext for this machine
+            new_access = contract_client.request_resource_machine_access(
+                new_token['machineToken'], name)
+        else:
+            new_access = entitlement
         try:
-            new_token = contract_client.request_machine_token_refresh(
-                machine_token=machine_token, contract_id=contract_id)
-        except util.UrlError:
-            return False
-    try:
-        for name, entitlement in sorted(cfg.entitlements.items()):
-            if entitlement['entitlement'].get('entitled'):
-                # Obtain each entitlement's accessContext for this machine
-                new_access = contract_client.request_resource_machine_access(
-                    new_token['machineToken'], name)
-            else:
-                new_access = entitlement
             process_entitlement_delta(
                 orig_entitlements.get(name, {}), new_access,
                 allow_enable=allow_enable)
-    except util.UrlError as e:
-        logging.error(
-            'Could not obtain updated contract information. %s', str(e))
-        return False
-    return True
+        except exceptions.UserFacingError as e:
```

Comment:
Should we be catching all exceptions here? If we don't then they'll bubble all the way out to the user and cause the process to exit 1, I think.

Reply:
(To be clear, I mean in an additional exception handler; we would probably want to log the exception and append a generic message to errors.)